### PR TITLE
Mothership Lab Vault: Floor Fixin' (Simulated -> Unsimulated) and Some Addons

### PR DESCRIPTION
**Pull request by Optimism333**
## Floor Maintenance ##
If you don't want to read the barely coherent ramblings of an amateur coder/mapper who stayed up too late, just skip to the next section for a tldr.

- If I'd known about the difference between unsimulated floors and simulated floors when I started mapping this vault, I probably could have saved myself a lot of work that I ended up having to do now.
- This pr attempts to correct an issue that was exacerbated by the fact that the hoverdisc drone mobs in the mothership lab explode when destroyed. That issue is that it was pretty easy to depressurize the vault by destroying the floor, accidentally or purposefully. I replaced as many simulated floors as I could with unsimulated floors, so hopefully this shouldn't happen much anymore.
- This has positive and negative effects. On the plus side for players, a space suit is optional once inside the vault, as the floor is a lot less likely to be breached. On the negative side, the ayy mobs (most of which aren't spess worthy) can't all be suffocated by blowing holes in the floor. I don't really see that second one as a downside though, as that's not a method of clearing I wanted to encourage for this vault.
- You may have noticed I used a lot of uncertain words: "hopefully" "shouldn't happen much" "a lot less likely". There's a lot of floor tiles in this vault. It's guaranteed that I missed at least a few. But if some clever player finds one that can STILL be depressurized and uses it to their advantage, good for them.

## TLDR ##
- I fixed the floor of the mothership lab so it doesn't get destroyed so easily anymore (as much as I could, anyways)
- I also made a few changes to the levels, mostly the entry level and the habitation deck, screenshots attached

### Nu Entry Level ###
mothershiplab
**Noteable Changes**
- Added explorer's barracks
- Added a little reward for someone who checks every nook and cranny

### Nu Habitation Deck ###
habitationdeck
**Noteable Changes**
- The old lab maintenance now functions more like an actual maintenance area, allowing for several backdoor entry points to certain rooms
- Added a NASA voidsuit to the old lab maintenance

[content] [vault]

:cl:
 * tweak: The mothership lab vault floors have been strengthened
 * rscadd: The mothership lab vault has a few new doors and rooms in select areas

### CODE REVIEW

**Comment by ValkyrieSkies** [comment]
Couldn't you have just reduced the strength of the hover drone explosion so that it couldn't breach the floor?

**Comment by Optimism333** [comment]
> Couldn't you have just reduced the strength of the hover drone explosion so that it couldn't breach the floor?

That would have been the easy way, but I consider that a feature of the mob that I'd rather keep.

**Comment by ValkyrieSkies** [comment]
I dunno man, using unsimmed tiles just doesn't seem great. One of the biggest things brought up on our Guide to Mapping is that Unsimmed and Simmed tiles should not be mixed.

**Comment by Optimism333** [comment]
> I dunno man, using unsimmed tiles just doesn't seem great. One of the biggest things brought up on our Guide to Mapping is that Unsimmed and Simmed tiles should not be mixed.

I suppose I'll wait and see if a mapper or coder who knows better than me tells me off, then. There's already been some of that in this vault prior to this pr, though. And my vault isn't the only one that does that either.

On some further thought, I'm pretty sure that guidance is intended for station maps. I could see how that could mess things up there, but vaults and away missions tend to have more leeway. Could be wrong, though.

**Comment by Eneocho** [comment]
Definitely against mixing simulated and unsimulated tiles, at least not to this extent, that just calls for gamebreaking issues. You could just make the explosion not breach but wider to compensate.

**Comment by Optimism333** [comment]
> Definitely against mixing simulated and unsimulated tiles, at least not to this extent, that just calls for gamebreaking issues. You could just make the explosion not breach but wider to compensate.

I don't think there even is that much mixing at this point. Most of the asteroid is unsimulated and so are most of the walls in the vault. After the floor change the only simulated tiles left would be some of the interior r-walls, and most of the porous rock walls.

But either way, if mixing is a big issue, can't I just fix that by converting most of the remaining simulated turfs to unsimulated ones? If I had to pick one or the other, I would rather have the breach-proof unsimulated floors for the vault. It's closer in line with how I actually wanted it to function from the beginning.

**Comment by Optimism333** [comment]
Hobo shack floors are now unsimulated. So are the remaining asteroid floors that I missed.

**Comment by Eneocho** [comment]
Not sure how feasible it is since I haven't really dabbled on mapping and mapcode, but you could make it so instead of a breach to space, it breaches to an asteroid tile, with the whole gas simulation it implies.

**Comment by Optimism333** [comment]
> Not sure how feasible it is since I haven't really dabbled on mapping and mapcode, but you could make it so instead of a breach to space, it breaches to an asteroid tile, with the whole gas simulation it implies.

Not entirely sure how to do that, but I'm pretty sure I'd have to do something very similar to what I've done here, but with an extra step. Which is to code a new piece of simulated flooring that fills the turf with asteroid when destroyed and then rip out all the floors and replace them with those tiles.

That would be cooler than what I've done here, but it'd basically accomplish the same goal as this pr. For now at least, I'd like to not do all the work over again if I don't have to.

**Comment by Kurfursten** [comment]
Yes you can do that, I added the technology for Snaxi years ago
